src/test/java/htsjdk/samtools/cram/structure/BlockTest.java

```diff
@@ -102,12 +103,54 @@ public void testExternalBlockRoundTrips() throws IOException {
         final byte[] uncompressedData = "A TEST STRING WITH REDUNDANCY AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA".getBytes();
         final byte[] compressedData = compressor.compress(uncompressedData);
 
-        final Block extBlock = new ExternalBlock(compressor.getMethod(), contentID, compressedData, uncompressedData.length);
+        final Block extBlock = Block.createExternalBlock(compressor.getMethod(), contentID, compressedData, uncompressedData.length);
 
         final Block rtBlock2 = roundTrip(extBlock, CramVersions.CRAM_v2_1);
         contentCheck(rtBlock2, uncompressedData, compressedData);
 
         final Block rtBlock3 = roundTrip(extBlock, CramVersions.CRAM_v3);
         contentCheck(rtBlock3, uncompressedData, compressedData);
     }
+
+    private class BlockTester extends Block {
+        private BlockTester(final BlockCompressionMethod compressionMethod,
+                           final BlockContentType contentType,
+                           final int contentId,
+                           final byte[] compressedContent,
+                           final int uncompressedLength) {
+            super(compressionMethod, contentType, contentId, compressedContent, uncompressedLength);
+        }
+    }
```

Comment:
If you moved this file into the block package I think you wouldn't need this ?

Reply:
Right!